trailblazer/store/filters/analyses_filters.py

```diff
@@ -23,6 +24,23 @@ def filter_analyses_by_entry_id(analyses: Query, analysis_id: int, **kwargs) ->
     return analyses.filter(Analysis.id == analysis_id)
 
 
+def filter_analyses_by_is_visible(analyses: Query, **kwargs) -> Query:
+    """Filter analyses by case when is visible is true."""
+    return analyses.filter(Analysis.is_visible.is_(True))
+
+
+def filter_analyses_by_search_term(analyses: Query, search_term: str, **kwargs) -> Query:
+    """Filter analyses by search term using multiple fields."""
+    return analyses.filter(
+        sqlalchemy.or_(
+            Analysis.family.ilike(f"%{search_term}%"),
+            Analysis.status.ilike(f"%{search_term}%"),
+            Analysis.data_analysis.ilike(f"%{search_term}%"),
+            Analysis.comment.ilike(f"%{search_term}%"),
+        )
```

Comment:
Not sure if this function is a tad too general in terms of what the allowed fields are it searches.
What is the end goal of this filter? Is it used for display purposes only ?
What if we have an Analysis.comment that contains a `data_analysis` that is not the same as the `Analysis.data_analysis` one. Could this lead to issues?

Reply:
Agreed, but needs some work in the FE to remodel this. Should not be done now. This function is only used in the FE for display ad very greedy searching so no real harm.